lib/trino-filesystem-s3/src/main/java/io/trino/filesystem/s3/S3FileSystemFactory.java

```diff
@@ -106,6 +106,10 @@ public S3FileSystemFactory(OpenTelemetry openTelemetry, S3FileSystemConfig confi
                     config.getHttpProxy()));
             httpClient.proxyConfiguration(ProxyConfiguration.builder()
                     .endpoint(endpoint)
+                    .username(config.getHttpProxyUsername())
```

Comment:
Curiosity question: Were you able to test these changes yourself?

Reply:
nope. just trusting the sdk :) seems straightforward